pkg/core/data/transactions/fields.go

```diff
@@ -288,6 +288,11 @@ type SecretKey struct {
 	B *Scalar `json:"b"`
 }
 
+// IsEmpty is used when passing the SecretKey by value
+func (s SecretKey) IsEmpty() bool {
+	return s.A == nil && s.B == nil
+}
+
 // MSecretKey copy the USecretKey from rusk to transactions datastruct
 func MSecretKey(r *rusk.SecretKey, n *SecretKey) {
 	r.A = new(rusk.Scalar)
```

Comment:
Should we have the `IsEmpty` check here as well, like in `MPublicKey`?

Reply:
In theory sounds good. However, the `SecretKey` is normally required. I will fix it but we might want to add some verification by the caller